xrpl/models/transactions/transaction.py

```diff
@@ -275,18 +275,28 @@ def to_dict(self: Self) -> Dict[str, Any]:
         """
         # we need to override this because transaction_type is using ``field``
         # which will not include the value in the objects __dict__
-        return {
+        prepared_dict = {
```

Comment:
```suggestion
        prepared_dict = {
            **super().to_dict(),
            "transaction_type": self.transaction_type.value,
            "flags": self._flags_to_int(),
        }

        if self.flags is None:
            del prepared_dict["flags"]
        return prepared_dict
```

This would remove the flags field only if it was not set initially. If it is set as 0 we would keep it as it is so that `blob` method generates the correct binary format and after this change there shouldn't be a need to make changes in core/binarycodec tests.

Reply:
I believe the expected behaviour of `from_dict` and `to_xrpl` should result in a minimalist transaction. 
If providing Flags filed with value of 0 in a dict that is used to create a transaction object, the `to_dict` and `to_xrpl` should remove the flag as it has no effect.

I added 2 tests to clarify. Let me know if you disagree

Reply:
@AvbrehtLuka xrpl.js SDK does not remove the Flags field if its 0 when signing and encoding the transaction. So, if we remove it in xrpl-py then the encoded transaction would be different (equally valid though) and we want to keep these libraries in sync. So, its better to keep `flags` as 0 if its explicitly present.

@mvadari Please pitch in if the above analysis don't make sense and its fine to have different `tx_blob` for both xrpl-py and xrpl.js for the same transaction. 
Also, on another note, do we consider this a breaking change since, after this PR gets released, the `encode` in xrpl-py will generate a different output as opposed to what we had previously. If so, do we refrain from merging this in main and merge this is another `release` branch that would be released with a major version bump?

Reply:
I would personally rather just ensure that the `flags_to_int` helper function returns `None` if `flags` is None, rather than add additional code for this check.

xrpl.js can also be checked to ensure it handles no `Flags` field properly.

`encode` from the binary codec isn't changing, personally I think this is more of a bugfix than a breaking change since it's all internal changes and no behavioral changes.

Reply:
The js lib does not really have the `from_dict` equivalent as far as I know. Even if we change  the `_flags_to_int` helper function to return None, the same problem persists, meaning if you generate a transaction from dict in which you have flags field explicitly set to `"flags": 0`, the `to_dict` method will return the  the object with flags set to 0 that has no effect. When creating a instance of frozen class, for instance a Payment instance with flags set to 0, the flags filed will not be present in the so generated dict. This is different to how the js lib handles the same example. I can see why you would prefer to keep the explicit `"flags": 0` even if it does nothing. 

Let me know:
1. Should the to_dict method have flags set to 0 if it is explicitly set to 0
2. Should we remove the flags filed if flags are set to zero and have no effect. 

As soon as you let me know I can update and add basic tests to ensure consistency in the future.

Reply:
Going one step further:
```suggestion
        prepared_dict = {
	            **super().to_dict(),
	            "transaction_type": self.transaction_type.value,
	        }
	
	        if flags := self._flags_to_int() is not None:
	            prepared_dict["flags"] = self._flags_to_int()
	        return prepared_dict
```